File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(std::vector<int>& nums) {
        std::unordered_set<int> set;
        set.reserve(nums.size());

        for (int num : nums) {
            if (set.find(num) != set.end()) {
                return true;
            }
            set.insert(num);
        }
        return false;
    }
};

File: Data Structures & Algorithms/house-robber/submission-0.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        dp = [0] * len(nums)

        for i, num in enumerate(nums):
            if i == 0:
                dp[i] = num
            elif i == 1:
                dp[i] = max(dp[i - 1], num)
            else:
                dp[i] = max(dp[i - 2] + num, dp[i - 1])
        
        return dp[-1]
